wmutils-opt: init at v1.0

vifino opened the pull request:
Hello!

###### Motivation for this change
wmutils-opt has some nice helpers that complement wmutils-core, but there was no package for that yet,
so I did it.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I hope that this is an acceptable PR. I have only used Nix for a few hours, so I am quite inexperienced.

rasendubi commented on pkgs/tools/X11/wmutils-opt/default.nix:
```diff
@@ -0,0 +1,22 @@
+  name = "wmutils-opt-${version}";
+  version = "1.0";
+
+  src = fetchurl {
```
You could use `fetchFromGitHub` as a shortcut:
```nix
fetchFromGitHub {
  owner = "wmutils";
  repo = "opt";
  rev = "v${version}";
  sha256 = "0mfhpja2l592i2a3j2l0lg65jhsdw1b57jnv2l0759ngpl4v82zk";
}

vifino replied:
That sounds good!
Should I add that in another commit or change the existing one?

rasendubi replied:
Squashing is preferred.

rasendubi commented on pkgs/tools/X11/wmutils-opt/default.nix:
```diff
@@ -0,0 +1,22 @@
+
+  installFlags = [ "PREFIX=$(out)" ];
+
+  meta = with stdenv.lib; {
```
Would you like to maintain the package?

vifino replied:
Sure! Isn't much work anyways, nor do they update often.

rasendubi replied:
Then, could you please add yourself as a maintainer in the `meta`? (Don't forget to also add a line to `lib/maintainers.nix`.)

rasendubi reviewed (commented):
Otherwise, looks good. Builds fine on NixOS with sandboxing.

vifino commented:
I changed the fetch directive to the `fetchFromGithub` one, added myself to the maintainers list and added the line `maintainers = with stdenv.lib.maintainers; [ vifino ];` in the meta. Anything else I forgot?

rasendubi commented:
Pushing your changes? :smile:

vifino commented:
Oh, almost forgot that one! ;)

rasendubi commented on pkgs/tools/X11/wmutils-opt/default.nix:
```diff
@@ -0,0 +1,25 @@
+    repo = "opt";
+    rev = "v${version}";
+    sha256 = "0mfhpja2l592i2a3j2l0lg65jhsdw1b57jnv2l0759ngpl4v82zk";
+  }
```
Missed semicolon.

vifino replied:
Dammit! So close!

rasendubi commented on pkgs/tools/X11/wmutils-opt/default.nix:
```diff
@@ -0,0 +1,25 @@
+    description = "Optional addons to wmutils";
+    homepage = https://github.com/wmutils/opt;
+    license = licenses.isc;
+    maintainers = with stdenv.lib.maintainers; [ vifino ];
```
You have already done `with stdenv.lib;`, so you could do `with maintainers;` here.

vifino replied:
Okay.

rasendubi commented on pkgs/tools/X11/wmutils-opt/default.nix:
```diff
@@ -0,0 +1,25 @@
+  name = "wmutils-opt-${version}";
+  version = "1.0";
+
+  src = fetchFromGitHub {
```
Ouch, we haven't imported `fetchFromGitHub`.

rasendubi commented on pkgs/tools/X11/wmutils-opt/default.nix:
```diff
@@ -0,0 +1,25 @@
+{ stdenv, fetchurl, libxcb }:
```
Let's replace `fetchurl` with `fetchFromGitHub`.

vifino commented:
There are only so many things left I can do wrong :smile:

rasendubi commented on pkgs/tools/X11/wmutils-opt/default.nix:
```diff
@@ -0,0 +1,25 @@
+    owner = "wmutils";
+    repo = "opt";
+    rev = "v${version}";
+    sha256 = "0mfhpja2l592i2a3j2l0lg65jhsdw1b57jnv2l0759ngpl4v82zk";
```
Yeah... hash is now wrong:
```
unpacking source archive /tmp/nix-build-opt-v1.0-src.drv-0/v1.0.tar.gz
output path ‘/nix/store/swar2bmcm82243cb74i0mln8s89s9l8s-opt-v1.0-src’ has r:sha256 hash ‘0gd05qsir1lnzfrbnfh08qwsryz7arwj20f886nqh41m87yqaljz’ when ‘0mfhpja2l592i2a3j2l0lg65jhsdw1b57jnv2l0759ngpl4v82zk’ was expected
```

Put `0gd05qsir1lnzfrbnfh08qwsryz7arwj20f886nqh41m87yqaljz` in there.

vifino committed:
wmutils-opt: init at v1.0

vifino commented:
:tada: :tada: :tada:

Thanks for suggesting these changes, now I know better, hopefully it won't be so much effort on your end! :smile:

rasendubi commented:
Yay! Thank you for your contribution and welcome to nixpkgs! :tada:

You could check your package builds before pushing changes. This could have saved us three review rounds. :smiley: 

Run the following command from the root of nixpkgs:
```sh
nix-build -A wmutils-opt
```

vifino commented:
I very much should have done that, have yet to set up a proper dev env for nix/nixos, so testing is hard at the moment. I have yet to get gnome-boxes to cooperate and create a bridge, I just cloned my fork to test it, which resulted in the PR obviously getting updated too.
